File: src/main/java/us/illyohs/bansdb/util/DBUtil.java
package us.illyohs.bansdb.util;

import javax.sql.DataSource;
import java.sql.Connection;
import java.sql.SQLException;
import java.util.List;

import org.spongepowered.api.Sponge;
import org.spongepowered.api.service.sql.SqlService;

public class DBUtil {

    private static SqlService sqlServ = null;
    private static DataSource ds      = null;

    public static void init() {

        sqlServ = Sponge.getServiceManager().provide(SqlService.class).get();
        String sqlURL = ""; //TODO: add a config
        try {
            ds = sqlServ.getDataSource(sqlURL);
            Connection connection = ds.getConnection();
            //TODO: Make a lib and make a builder pattern for the sake of my lazyness;
            connection.prepareStatement(
                    "CREATE TABLE IF NOT EXISTS BANS_DB ("+
                            "`banneeUuid` CHARACTER(36) NOT NULL" + //Bannee's uuid
                            "`bannerUuid` CHARACTER(36) NOT NULL" + //Banner's uuid
                            "`isBanned` SMALLINT NOT NULL DEFAULT '0'," + //Is the Player banned
                            "`isMuted` SMALLINT NOT NULL DEFAULT '0'" + //Is the player muted
                            "`reason` MEDIUMTEXT NOT NULL DEFAULT ''" + //
                            ");"
            );

        } catch (SQLException e) {
            e.printStackTrace();

        }
    }

    public void checkDB() {

    }

    public boolean isPlayerBanned() {
        return false;
    }

    public boolean isPlayerMuted() {
        return false;
    }

    public List<BansMeta> getBans() {
        return null;
    }

    public List<BansMeta> getMutes() {
        return null;
    }

}

File: src/main/scala/us/illyohs/bansdb/util/DBUtil.scala
package us.illyohs.bansdb.util

import javax.sql.DataSource
import java.sql.Connection
import java.sql.SQLException
import java.sql.Statement
import java.util.UUID

import org.spongepowered.api.Sponge
import org.spongepowered.api.service.sql.SqlService
import us.illyohs.bansdb.util.ConfigUtil._


object DBUtil {
  private var sqlServ: SqlService = null
  private var ds: DataSource = null

  def execute(execute: String): Unit = {
    sqlServ = Sponge.getServiceManager.provide(classOf[SqlService]).get()
    val URL: String = "jdbc:mysql://" + getURL + ":" + getPort + "/" + "?user=" + getUsername + "&password=" + getPass
    try {
      ds = sqlServ.getDataSource(URL)
      val connection: Connection = ds.getConnection
      val statement: Statement = connection.createStatement
      statement.execute(execute)
      statement.close
      connection.close
    }
    catch {
      case e: SQLException => e.printStackTrace
    }
  }

  def init: Unit = {
    execute("CREATE TABLE IF NOT EXISTS WARN_DB (warnee TEXT, sender TEXT, warnings SMALLINT, reason TEXT)")
    execute("CREATE TABLE IF NOT EXISTS BAN_DB (banee TEXT, sender TEXT, reason)")
  }


  def getWarning: Int = {
    null
  }

  def isPlayerBanned(uUID: UUID): Boolean = {
    return false
  }

  def isPlayerMuted(): Boolean = {
    return false
  }
}